### PR TITLE
clean up http Requests and WSGI layer, this allow non json route to h…

### PR DESCRIPTION
**Pull request by antonylesuisse**
…andle http requests

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by antonylesuisse** [comment]
WIP, not yet executed but almost everything is done.

File: odoo/http.py [hunk 1]
```diff
@@ -420,451 +616,262 @@ def validate_csrf(self, csrf):
...
+        self.rpc_debug_post(t0, result)
 
-    In case of name conflict, routing parameters have priority.
+        return self._json_response(result)
 
-    The handler method's result can be:
+    #------------------------------------------------------
+    # Entry point
+    #------------------------------------------------------
+    def dispatch(self, endpoint, args, auth='none'):
```
Comment (std-odoo) [inline odoo/http.py]: Where is it called ?

**Comment by seb-odoo** [comment]
@antonylesuisse do you still need this PR?

**Comment by rdeodoo** [comment]
Since there were httpocalypse and websocket refactoring since, I guess this can be closed?
If still (partly) needed, this can be reopened

**Pull request closed without merging**